pkg/kubelet/dockershim/convert.go

```diff
@@ -141,6 +144,9 @@ func toRuntimeAPISandboxState(state string) runtimeApi.PodSandBoxState {
 
 func toRuntimeAPISandbox(c *dockertypes.Container) (*runtimeApi.PodSandbox, error) {
 	state := toRuntimeAPISandboxState(c.Status)
+	if len(c.Names) == 0 {
+		return nil, fmt.Errorf("unexpected empty sandbox name: %+v", c)
```

Comment:
> so never restart a reastart:never pod then?

It's possible that kubelet will restart a restart:never pod in this case, since it cannot recognize the container anymore. 
The last time I encountered this bug (https://github.com/kubernetes/kubernetes/issues/21085#issuecomment-186439716), it was with docker v1.9. Haven't seen it happen on a newer docker version.
This stackoverflow [question/answer](http://stackoverflow.com/questions/30550472/docker-container-with-status-dead-after-consul-healthcheck-runs) has more information.